Assets/Scripts/Augment/AugmentImplementations/RopeBody.cs

```diff
@@ -73,48 +91,150 @@ private void PullingWire()
     {
-        isWired = false;
-        rope.enabled = false;
-        gunController.stats.Ammo = 0;
-        plugAnchor.gameObject.SetActive(false);
-        audioSource.Play();
+        if (isWired == false)
+            return;
+        RemoveRope();
     }
 
     private void CheckForWirePlanting(GunStats stats)
     {
-        if (isWired || movement.StateIsAir)
+        if (isWired || isThrowing || !canThrow)
             return;
-        plugAnchor.transform.position = gunController.Player.transform.position;
-        movement.CanMove = false;
-        movement.CanLook = false;
+
         animator.SetTrigger("Plant");
-        isWired = true;
-        rope.ResetRope(plugAnchor.WireOrigin);
+        isThrowing = true;
     }
 
     // Called by animator
     public void ActivateWire()
     {
         if (!plugAnchor)
             return;
-        rope.enabled = true;
-        plugAnchor.gameObject.SetActive(true);
-        movement.CanMove = true;
-        movement.CanLook = true;
-        gunController.stats.Ammo = gunController.stats.MagazineSize;
+        if (Physics.Raycast(gunController.Player.inputManager.transform.position, gunController.Player.inputManager.transform.forward, out RaycastHit hit, ropeLength, ignoreLayer))
+        {
+            plugAnchor.transform.position = gunController.Player.transform.position;
+            plugAnchor.transform.forward = hit.normal;
+            rope.enabled = true;
+            canThrow = false;
+            LeanTween.value(gameObject, SetThrowValue, 0f, ropeLength, 0.02f * ropeLength);
+            plugAnchor.gameObject.LeanMove(hit.point, 0.02f * hit.distance)
+                .setOnComplete(() =>
+                {
+                    rope.ResetRope(plugAnchor.WireOrigin);
+                    isThrowing = false;
+                    isWired = true;
+                    plugAnchor.SetPulseStrength(0.85f);
+                    gunController.stats.Ammo = gunController.stats.MagazineSize;
+                });
+            rope.ResetRope(plugAnchor.WireOrigin);
+            plugAnchor.gameObject.SetActive(true);
+        }
+        else
+        {
+            plugAnchor.SetPulseStrength(0);
+            plugAnchor.transform.position = gunController.Player.transform.position;
+            plugAnchor.transform.forward = gunController.Player.inputManager.transform.forward;
+            rope.enabled = true;
+            plugAnchor.gameObject.SetActive(true);
+            AnimateFullPullBack();
+        }
+    }
+
+    public void PlayThrowAudio()
+    {
+        audioSource.clip = plugThrow;
+        audioSource.Play();
+    }
+
+    private void AnimateFullPullBack()
+    {
+        LeanTween.value(gameObject, SetThrowValue, 0f, ropeLength, 0.02f * ropeLength)
+            .setEaseOutElastic();
+
+        plugAnchor.gameObject.LeanMove(gunController.Player.inputManager.transform.position + gunController.Player.inputManager.transform.forward * ropeLength, 0.02f * ropeLength)
```

Comment:
This may look a little odd if the player moves, as far as I can tell?

Reply:
The cause was from another place, have fixed it now 👍

Reply:
Seems that way 👍